client/cli/src/params/network_params.rs

```diff
@@ -48,62 +48,62 @@ pub struct NetworkParams {
 	/// In particular, if you are a validator your node might still connect to other
 	/// validator nodes and collator nodes regardless of whether they are defined as
 	/// reserved nodes.
-	#[clap(long)]
+	#[arg(long)]
 	pub reserved_only: bool,
 
 	/// The public address that other nodes will use to connect to it.
 	/// This can be used if there's a proxy in front of this node.
-	#[clap(long, value_name = "PUBLIC_ADDR", multiple_values(true))]
+	#[arg(long, value_name = "PUBLIC_ADDR", num_args = 0..)]
 	pub public_addr: Vec<Multiaddr>,
 
 	/// Listen on this multiaddress.
 	///
 	/// By default:
 	/// If `--validator` is passed: `/ip4/0.0.0.0/tcp/<port>` and `/ip6/[::]/tcp/<port>`.
 	/// Otherwise: `/ip4/0.0.0.0/tcp/<port>/ws` and `/ip6/[::]/tcp/<port>/ws`.
-	#[clap(long, value_name = "LISTEN_ADDR", multiple_values(true))]
+	#[arg(long, value_name = "LISTEN_ADDR", num_args = 0..)]
 	pub listen_addr: Vec<Multiaddr>,
 
 	/// Specify p2p protocol TCP port.
-	#[clap(long, value_name = "PORT", conflicts_with_all = &[ "listen-addr" ])]
+	#[arg(long, value_name = "PORT", conflicts_with_all = &[ "listen_addr" ])]
 	pub port: Option<u16>,
 
 	/// Always forbid connecting to private IPv4 addresses (as specified in
 	/// [RFC1918](https://tools.ietf.org/html/rfc1918)), unless the address was passed with
 	/// `--reserved-nodes` or `--bootnodes`. Enabled by default for chains marked as "live" in
 	/// their chain specifications.
-	#[clap(long, conflicts_with_all = &["allow-private-ipv4"])]
+	#[arg(long, conflicts_with_all = &["allow_private_ipv4"])]
 	pub no_private_ipv4: bool,
 
 	/// Always accept connecting to private IPv4 addresses (as specified in
 	/// [RFC1918](https://tools.ietf.org/html/rfc1918)). Enabled by default for chains marked as
 	/// "local" in their chain specifications, or when `--dev` is passed.
-	#[clap(long, conflicts_with_all = &["no-private-ipv4"])]
+	#[arg(long, conflicts_with_all = &["no_private_ipv4"])]
 	pub allow_private_ipv4: bool,
 
 	/// Specify the number of outgoing connections we're trying to maintain.
-	#[clap(long, value_name = "COUNT", default_value = "25")]
+	#[arg(long, value_name = "COUNT", default_value = "25")]
```

Comment:
If we're adjusting to best practices: these could use the stong-typed `default_value_t`, I think? Same for any other arguments (especially enums).

Reply:
This is a good point, transformed this were possible